// @adyen/api-library/lib/src/typings/configurationWebhooks/cardOrderItemDeliveryStatus.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardOrderItemDeliveryStatus = void 0;
class CardOrderItemDeliveryStatus {
    static getAttributeTypeMap() {
        return CardOrderItemDeliveryStatus.attributeTypeMap;
    }
}
exports.CardOrderItemDeliveryStatus = CardOrderItemDeliveryStatus;
CardOrderItemDeliveryStatus.discriminator = undefined;
CardOrderItemDeliveryStatus.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "CardOrderItemDeliveryStatus.StatusEnum"
    },
    {
        "name": "statusError",
        "baseName": "statusError",
        "type": "string"
    },
    {
        "name": "statusErrorMessage",
        "baseName": "statusErrorMessage",
        "type": "string"
    },
    {
        "name": "trackingNumber",
        "baseName": "trackingNumber",
        "type": "string"
    }
];
(function (CardOrderItemDeliveryStatus) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Created"] = "created";
        StatusEnum["Delivered"] = "delivered";
        StatusEnum["Processing"] = "processing";
        StatusEnum["Produced"] = "produced";
        StatusEnum["Rejected"] = "rejected";
        StatusEnum["Shipped"] = "shipped";
        StatusEnum["Unknown"] = "unknown";
    })(StatusEnum = CardOrderItemDeliveryStatus.StatusEnum || (CardOrderItemDeliveryStatus.StatusEnum = {}));
})(CardOrderItemDeliveryStatus = exports.CardOrderItemDeliveryStatus || (exports.CardOrderItemDeliveryStatus = {}));
