// @adyen/api-library/lib/src/typings/configurationWebhooks/cronSweepSchedule.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CronSweepSchedule = void 0;
class CronSweepSchedule {
    static getAttributeTypeMap() {
        return CronSweepSchedule.attributeTypeMap;
    }
}
exports.CronSweepSchedule = CronSweepSchedule;
CronSweepSchedule.discriminator = undefined;
CronSweepSchedule.attributeTypeMap = [
    {
        "name": "cronExpression",
        "baseName": "cronExpression",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "CronSweepSchedule.TypeEnum"
    }
];
(function (CronSweepSchedule) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Daily"] = "daily";
        TypeEnum["Weekly"] = "weekly";
        TypeEnum["Monthly"] = "monthly";
        TypeEnum["Balance"] = "balance";
        TypeEnum["Cron"] = "cron";
    })(TypeEnum = CronSweepSchedule.TypeEnum || (CronSweepSchedule.TypeEnum = {}));
})(CronSweepSchedule = exports.CronSweepSchedule || (exports.CronSweepSchedule = {}));
