// @adyen/api-library/lib/src/typings/configurationWebhooks/sweepConfiguration.js
"use strict";
/*
 * The version of the OpenAPI document: v1
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit this class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SweepConfiguration = void 0;
class SweepConfiguration {
    static getAttributeTypeMap() {
        return SweepConfiguration.attributeTypeMap;
    }
}
exports.SweepConfiguration = SweepConfiguration;
SweepConfiguration.discriminator = undefined;
SweepConfiguration.attributeTypeMap = [
    {
        "name": "balanceAccountId",
        "baseName": "balanceAccountId",
        "type": "string"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "merchantAccount",
        "baseName": "merchantAccount",
        "type": "string"
    },
    {
        "name": "schedule",
        "baseName": "schedule",
        "type": "CronSweepSchedule | SweepSchedule"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "SweepConfiguration.StatusEnum"
    },
    {
        "name": "sweepAmount",
        "baseName": "sweepAmount",
        "type": "Amount"
    },
    {
        "name": "targetAmount",
        "baseName": "targetAmount",
        "type": "Amount"
    },
    {
        "name": "transferInstrumentId",
        "baseName": "transferInstrumentId",
        "type": "string"
    },
    {
        "name": "triggerAmount",
        "baseName": "triggerAmount",
        "type": "Amount"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "SweepConfiguration.TypeEnum"
    }
];
(function (SweepConfiguration) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum["Active"] = "active";
        StatusEnum["Inactive"] = "inactive";
    })(StatusEnum = SweepConfiguration.StatusEnum || (SweepConfiguration.StatusEnum = {}));
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum["Pull"] = "pull";
        TypeEnum["Push"] = "push";
    })(TypeEnum = SweepConfiguration.TypeEnum || (SweepConfiguration.TypeEnum = {}));
})(SweepConfiguration = exports.SweepConfiguration || (exports.SweepConfiguration = {}));
